Add files via upload

Belyaash opened the pull request.

Belyaash committed:
Add files via upload

Belyaash committed:
Add files via upload

sinkapoy commented on Lab1/Model/SudokuAppModel/ISudokuAppModel.py:
```diff
@@ -0,0 +1,37 @@
+from Model.cell import Cell
+
+
+class ISudokuAppModel:
```
Зачем делать отдельно интерфейс, если полиморфизм не используется, а модель будет синглтоном?

Belyaash replied:
Разве не стоит оставлять абстракцию при взаимодействии между классами? Это как минимум позволит быстро и безболезненно заменить модель (если я например найду БД с сформированными полями судоку)

sinkapoy commented on Lab1/Model/SudokuAppModel/sudokuAppModel.py:
```diff
@@ -0,0 +1,49 @@
+
+    def is_player_win(self) -> bool:
+        for i in range(9):
+            for j in range(9):
```
O(n**2) - стоит добавить кеширование (искать по секторам и помечать готовые секторы) или эвристики. наверняка есть готовые решения

sinkapoy commented on Lab1/Model/SudokuAppModel/sudokuAppModel.py:
```diff
@@ -0,0 +1,49 @@
+    def is_game_grid_filled(self) -> bool:
+        for i in range(9):
+            for j in range(9):
+                if self.game_grid[i][j].num == 0:
```
мб через multithreading попробовать?

sinkapoy commented on Lab1/Model/SudokuGameGridFactory/sudokuGameGridFactory.py:
```diff
@@ -0,0 +1,67 @@
+
+
+class SudokuGameGridFactory(ISudokuGameGridFactory):
+    difficulty: int = None
```
нарушение типизации для атомарных типов, я бы 0 поставил

sinkapoy commented on Lab1/saver.py:
```diff
@@ -0,0 +1,15 @@
+import shelve
```
👍

sinkapoy commented on Lab1/main.py:
```diff
@@ -0,0 +1,9 @@
+
+from View.view_app import App
+
+if __name__ == '__main__':
```
нужна кнопка для решения судоку в дев-моде, потому что тратить полчаса на проверку сейвов мне некогда)

sinkapoy replied:
для разных уровней сложности нужны свои рекорды, потому что я могу закончить на легком, а потом переставить на сложный

sinkapoy replied:
выбор сложности должен быть до генерации игрового поля

sinkapoy replied:
Общая оценка: в целом хорошо, но не совсем понятен выбранный архитектурный шаблон - если это MVVM, то стоит явно обозначить, где находится VM; если это MVP, то стоит явно презентера выделить; если выбран свой подход, то стоит его в readme описать и дать аргументацию, почему выбран. Также стоит, наверное, отполировать флоу игрока: он запускает игру, хочет выбрать уровень или посмотреть рекорды на разных уровнях сложности, хочет сразу начать играть и т.д. (вообще работа геймдизайнера, но мы тут сами по себе). Также нет резиновости интерфейса: на 4к экране может быть очень мелко.

Belyaash replied:
> для разных уровней сложности нужны свои рекорды, потому что я могу закончить на легком, а потом переставить на сложный

Рекорды для каждого уровня уникальны. При смене уровня сложности автоматически создаётся новая игра

sinkapoy commented on Lab1/Model/SudokuGridFactory/sudokuGridFactory.py:
```diff
@@ -0,0 +1,74 @@
+        return self.__solved_grid
+
+    def __new_solved_grid(self) -> None:
+        self.__solved_grid = [[7, 1, 5, 3, 9, 8, 4, 6, 2],
```
это зачем?

Belyaash replied:
Класс больше не используется, забыл его удалить

sinkapoy commented on Lab1/Model/SudokuGridFactory/sudokuGridFactory.py:
```diff
@@ -0,0 +1,74 @@
+                              [2, 3, 9, 4, 8, 5, 6, 1, 7]]
+
+    def __shuffle_sudoku_grid(self, shuffles: int = 100) -> None:
+        mix_func = ['self._transposing()',
```
можно сделать отдельный объект и через dir выдрать ф-ии или сделать словарем. Хардкод строками - плохой подход, потому что при первом же переименовании прога посыпется

Belyaash replied:
Класс больше не используется, забыл его удалить

sinkapoy commented on Lab1/Model/SudokuSolver/sudokuSolver.py:
```diff
@@ -0,0 +1,102 @@
+
+        for row in range(0, 9):
+            for col in range(0, 9):
+                if self.__grid[row][col] == 0:
```
в 3 раз натыкаюсь на однотипный код, возможно, можно вынести в утилиты?

sinkapoy commented on Lab1/Model/SudokuSolver/sudokuSolver.py:
```diff
@@ -0,0 +1,102 @@
+                    self.__col_of_unfilled_cell = col
+                    return
+
+    @staticmethod
```
нет необходимости, т.к. используется только экземплярми класса

sinkapoy commented on Lab1/Model/SudokuSolver/sudokuSolver.py:
```diff
@@ -0,0 +1,102 @@
+        else:
+            return False
+
+    @staticmethod
```
то же самое

sinkapoy commented on Lab1/View/main_window.py:
```diff
@@ -0,0 +1,174 @@
+        Win dialog
+        :return:
+        """
+        win_dialog = QtWidgets.QErrorMessage(self)
```
нельзя сделать оверлеем в том же окне?

Belyaash committed:
Изменил названия файлов под MVVM паттерн.
Добалена тестировочная кнопка, позволяющая заполнить таблицу.
Убрана часть недочётов

Belyaash committed:
Изменил названия файлов под MVVM паттерн.
Добалена тестировочная кнопка, позволяющая заполнить таблицу.
Убрана часть недочётов

Belyaash committed:
Багфикс

Belyaash committed:
Delete Lab1/GUI/__pycache__ directory

Belyaash committed:
Delete Lab1/View/__pycache__ directory

Belyaash committed:
Delete Lab1/Model/__pycache__ directory

Belyaash committed:
Delete Lab1/Model/SudokuAppModel/__pycache__ directory

Belyaash committed:
Delete Lab1/Model/SudokuGameGridFactory/__pycache__ directory

Belyaash committed:
Delete Lab1/Model/SudokuGridFactory/__pycache__ directory

Belyaash committed:
Delete Lab1/Model/SudokuSolver/__pycache__ directory

Belyaash committed:
Выбор сложности теперь производится на старте

Belyaash committed:
Избавился от случаев нарушения типизации

Belyaash committed:
Немного изменил дизайн

Belyaash committed:
numpy.array вместь двумерных листов

Belyaash committed:
numpy.array вместь двумерных листов

Belyaash committed:
numpy.array вместь двумерных листов

Belyaash committed:
Полная замена листов на массивы, прирост производительности есть, но небольшой

Belyaash committed:
Небольшая оптимизация

Belyaash committed:
Небольшая оптимизация

Belyaash committed:
Небольшая оптимизация

Belyaash committed:
Новая реализация SudokuSolver'a. Асимптодика улучшилась

Belyaash committed:
Новая реализация SudokuSolver'a. Асимптодика улучшилась

Belyaash committed:
Чистка мусора

Belyaash committed:
Оптимизировал скорость

Belyaash committed:
Кусочек от генерации кастомных размеров

Belyaash committed:
Добавлена поддержка разных форматов судоку(тестил 2*2 и 4*4) в модель

Belyaash committed:
Сообщение о победе теперь выводится на том же окне, что и игра

Belyaash committed:
Выбор размеров поля от желания пользователя

Belyaash committed:
Интерфейс улучшили

Belyaash committed:
Интерфейс подшлифовал

Belyaash committed:
Небольшой рефакторинг вьюшек.

Рефакторинг viewmodel, код теперь более легко расширяем и более читабелен. При смене настроек сложности и размеров блока игра не создаётся заного автоматически, а создаётся при создании новой игры
